nomad/state/state_store.go

```diff
@@ -2421,21 +2421,21 @@ func (s *StateStore) UpsertCSIVolume(index uint64, volumes []*structs.CSIVolume)
 		}
 		if obj != nil {
 			// Allow some properties of a volume to be updated in place, but
-			// prevent accidentally overwriting important properties, or
-			// overwriting a volume in use
+			// prevent accidentally overwriting important properties.
 			old := obj.(*structs.CSIVolume)
 			if old.ExternalID != v.ExternalID ||
 				old.PluginID != v.PluginID ||
 				old.Provider != v.Provider {
 				return fmt.Errorf("volume identity cannot be updated: %s", v.ID)
 			}
-			s.CSIVolumeDenormalize(nil, old.Copy())
-			if old.InUse() {
-				return fmt.Errorf("volume cannot be updated while in use")
-			}
 
-			v.CreateIndex = old.CreateIndex
+			// Update fields that are safe to change while volume is being used.
+			if err := old.UpdateSafeFields(v); err != nil {
+				return fmt.Errorf("unable to update in-use volume: %w", err)
+			}
+			*v = *old
```

Comment:
Both objects are pointers, and you're just looking to rebind the value, so there's no need to do this.

```suggestion
			v = old
```